From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Wed, 26 Feb 2020 15:19:17 -0800
Subject: [PATCH] Use default, hardcoded config files with Include statements.

[2021-03-29]: Provide new conf files instead, to be renamed via
build_prepend. This avoids the need to rebase the patch for every conf
file change.
---
 conf/clr-zabbix_agentd.conf   | 4 ++++
 conf/clr-zabbix_server.conf   | 4 ++++
 src/zabbix_agent/Makefile.am  | 6 +++---
 src/zabbix_server/Makefile.am | 6 +++---
 4 files changed, 14 insertions(+), 6 deletions(-)
 create mode 100644 conf/clr-zabbix_agentd.conf
 create mode 100644 conf/clr-zabbix_server.conf

diff --git a/conf/clr-zabbix_agentd.conf b/conf/clr-zabbix_agentd.conf
new file mode 100644
index 0000000..e54e47b
--- /dev/null
+++ b/conf/clr-zabbix_agentd.conf
@@ -0,0 +1,4 @@
+PidFile=/run/zabbix/zabbix_agentd.pid
+LogType=system
+ListenIP=127.0.0.1
+Include=/etc/zabbix_agentd.conf
diff --git a/conf/clr-zabbix_server.conf b/conf/clr-zabbix_server.conf
new file mode 100644
index 0000000..e6c3efa
--- /dev/null
+++ b/conf/clr-zabbix_server.conf
@@ -0,0 +1,4 @@
+LogType=system
+PidFile=/run/zabbix/zabbix_server.pid
+ListenIP=127.0.0.1
+Include=/etc/zabbix_server.conf.d/*.conf
diff --git a/src/zabbix_agent/Makefile.am b/src/zabbix_agent/Makefile.am
index 86c188a..184f120 100644
--- a/src/zabbix_agent/Makefile.am
+++ b/src/zabbix_agent/Makefile.am
@@ -77,11 +77,11 @@ zabbix_agentd_LDFLAGS = $(AGENT_LDFLAGS)
 
 zabbix_agentd_CFLAGS = \
 	-DZABBIX_DAEMON \
-	-DDEFAULT_CONFIG_FILE="\"$(AGENT_CONFIG_FILE)\"" \
+	-DDEFAULT_CONFIG_FILE="\"/usr/share/defaults/zabbix/zabbix_agentd.conf\"" \
 	-DDEFAULT_LOAD_MODULE_PATH="\"$(LOAD_MODULE_PATH)\"" \
 	$(TLS_CFLAGS)
 
 install-data-hook:
-	$(MKDIR_P) "$(DESTDIR)$(AGENT_CONFIG_FILE).d"
+	$(MKDIR_P) "$(DESTDIR)/usr/share/defaults/zabbix"
 	$(MKDIR_P) "$(DESTDIR)$(LOAD_MODULE_PATH)"
-	test -f "$(DESTDIR)$(AGENT_CONFIG_FILE)" || cp "../../conf/zabbix_agentd.conf" "$(DESTDIR)$(AGENT_CONFIG_FILE)"
+	test -f "$(DESTDIR)$(AGENT_CONFIG_FILE)" || cp "../../conf/zabbix_agentd.conf" "$(DESTDIR)/usr/share/defaults/zabbix/zabbix_agentd.conf"
diff --git a/src/zabbix_server/Makefile.am b/src/zabbix_server/Makefile.am
index bb8e767..824ea84 100644
--- a/src/zabbix_server/Makefile.am
+++ b/src/zabbix_server/Makefile.am
@@ -155,15 +155,15 @@ zabbix_server_LDFLAGS = $(SERVER_LDFLAGS)
 zabbix_server_CFLAGS = \
 	-DDEFAULT_SSL_CERT_LOCATION="\"$(CURL_SSL_CERT_LOCATION)\"" \
 	-DDEFAULT_SSL_KEY_LOCATION="\"$(CURL_SSL_KEY_LOCATION)\"" \
-	-DDEFAULT_CONFIG_FILE="\"$(SERVER_CONFIG_FILE)\"" \
+	-DDEFAULT_CONFIG_FILE="\"/usr/share/defaults/zabbix/zabbix_server.conf\"" \
 	-DDEFAULT_EXTERNAL_SCRIPTS_PATH="\"$(EXTERNAL_SCRIPTS_PATH)\"" \
 	-DDEFAULT_ALERT_SCRIPTS_PATH="\"$(ALERT_SCRIPTS_PATH)\"" \
 	-DDEFAULT_LOAD_MODULE_PATH="\"$(LOAD_MODULE_PATH)\"" \
 	$(TLS_CFLAGS)
 
 install-data-hook:
-	$(MKDIR_P) "$(DESTDIR)$(SERVER_CONFIG_FILE).d"
+	$(MKDIR_P) "$(DESTDIR)/usr/share/defaults/zabbix"
 	$(MKDIR_P) "$(DESTDIR)$(EXTERNAL_SCRIPTS_PATH)"
 	$(MKDIR_P) "$(DESTDIR)$(ALERT_SCRIPTS_PATH)"
 	$(MKDIR_P) "$(DESTDIR)$(LOAD_MODULE_PATH)"
-	test -f "$(DESTDIR)$(SERVER_CONFIG_FILE)" || cp "../../conf/zabbix_server.conf" "$(DESTDIR)$(SERVER_CONFIG_FILE)"
+	test -f "$(DESTDIR)$(SERVER_CONFIG_FILE)" || cp "../../conf/zabbix_server.conf" "$(DESTDIR)/usr/share/defaults/zabbix/zabbix_server.conf"
